cmd/client_vanilla.go

```diff
@@ -0,0 +1,22 @@
+//go:build !openshift
+// +build !openshift
```

Comment:
I am happy to see that this is where it should be: `cmd` with a reference to an implementation-specific package.  This is the same as can be used for all future cases where flavor-specific code is necessary, not just for OpenShift.

My concern with this approach is that it results in a binary that is OpenShift specific.  This is not necessary, because the core `func` logic is agnostic if cluster flavor.

Is there not a way we can detect the cluster flavor at runtime?

Reply:
+1 here.. I would be more in favour of runtime detection with "standard" mechanisms, that we can reuse for other environments.